pkg/restic/backupper.go

```diff
@@ -84,39 +82,14 @@ func resultsKey(ns, name string) string {
 	return fmt.Sprintf("%s/%s", ns, name)
 }
 
-func getRepo(repoLister arkv1listers.ResticRepositoryLister, ns, name string) (*arkv1api.ResticRepository, error) {
-	repo, err := repoLister.ResticRepositories(ns).Get(name)
-	if apierrors.IsNotFound(err) {
-		return nil, errors.Wrapf(err, "restic repository not found")
-	}
-	if err != nil {
-		return nil, errors.Wrapf(err, "error getting restic repository")
-	}
-
-	return repo, nil
-}
-
-func getReadyRepo(repoLister arkv1listers.ResticRepositoryLister, ns, name string) (*arkv1api.ResticRepository, error) {
-	repo, err := getRepo(repoLister, ns, name)
-	if err != nil {
-		return nil, err
-	}
-
-	if repo.Status.Phase != arkv1api.ResticRepositoryPhaseReady {
-		return nil, errors.New("restic repository not ready")
-	}
-
-	return repo, nil
-}
-
 func (b *backupper) BackupPodVolumes(backup *arkv1api.Backup, pod *corev1api.Pod, log logrus.FieldLogger) (map[string]string, []error) {
 	// get volumes to backup from pod's annotations
 	volumesToBackup := GetVolumesToBackup(pod)
 	if len(volumesToBackup) == 0 {
 		return nil, nil
 	}
 
-	repo, err := getReadyRepo(b.repoLister, backup.Namespace, pod.Namespace)
+	repo, err := b.repoEnsurer.EnsureRepo(b.ctx, backup.Namespace, pod.Namespace)
 	if err != nil {
 		return nil, []error{err}
 	}
```

Comment:
This function is executed in the daemonset, correct?

I ran into an issue yesterday that was partially my fault, but probably warrants a log message. If the daemonset exists, but no pods in it are ready, the restic backup process will still proceed and then hang waiting for the daemonset pod to finish working. In my testing, it waited for over 5 minutes and couldn't be stopped with ctrl-c, only a kill -9.

I can probably replicate if needed to walk through this.